### PR TITLE
add year to xLabel in aggregate charts

### PR DESCRIPTION
**Pull request by austinjpaul**
Adding years to label the X-axis of our aggregate chart (so that screenshots on twitter can get full context).
[anyone can review-- i'm just picking people arbitrarily]

image

### CODE REVIEW

File: src/charts/line.js [hunk 1]
```diff
@@ -31,6 +31,8 @@ const prettyDate = (dateString, with_dow) => {
                         options);
 }
 
+const dateYear = (dateString) => {return dateString.split("-")[0]}
```
Comment (idreyn) [inline src/charts/line.js:34]: May as well drop the brackets and the `return` keyword:

```ts
const dateYear = (dateString) => dateString.split("-")[0];
```

File: src/charts/line.js [hunk 2]
```diff
@@ -317,7 +319,11 @@ const AggregateByDate = (props) => {
       xMin={new Date(`${props.startDate}T00:00:00`)}
       xMax={new Date(`${props.endDate}T00:00:00`)}
       fillColor={"rgba(191,200,214,0.5)"}
-      xLabel=""
+      xLabel={
+        dateYear(props.startDate) === dateYear(props.endDate) ?
```
Comment (idreyn) [inline src/charts/line.js:323]: $0.02 — we might as well store these in variables instead of calling `dateYear` on the same value several times.

**Review by idreyn — APPROVED** [review]
LGTM, this will be really helpful!

File: src/charts/line.js (hunk 2 above)
Comment (austinjpaul) [reply src/charts/line.js:323]: that's what I wanted to do, but I can't figure out how to put anything more complicated than a simple expression in the `props={}` section. Am I being silly? Should I just pull the whole thing into a `function dateLabel(d1, d2)`?

**Pull request merged**